__tests__/app.test.js

```diff
@@ -221,7 +220,257 @@ describe("POST /api/articles/:article_id/comments", () => {
+    return request(app)
+      .get("/api/articles/2?comment_count=false")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 2,
+          title: 'Sony Vaio; or, The Laptop',
+          topic: 'mitch',
+          author: 'icellusedkars',
+          body: expect.any(String),
+          created_at: "2020-10-16T05:03:00.000Z",
+          votes: 0,
+          article_img_url: expect.any(String),
+        });
+        expect(body.article.comment_count).toBeUndefined();
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles/2")
+          .expect(200);
+      })
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: 2,
+          title: 'Sony Vaio; or, The Laptop',
+          topic: 'mitch',
+          author: 'icellusedkars',
+          body: expect.any(String),
+          created_at: "2020-10-16T05:03:00.000Z",
+          votes: 0,
+          article_img_url: expect.any(String),
+        });
+        expect(body.article.comment_count).toBeUndefined();
+      });
+  });
+});
+
+describe("DELETE /api/comments/:comment_id", () => {
+  test("DELETE 204: deletes a comment and responds with no content", () => {
+    let testedCommentCount;
+    let commentId;
+    return request(app)
+      .get("/api/articles/1?comment_count=true")
+      .expect(200)
+      .then(({ body }) => {
+          testedCommentCount = body.article.comment_count;
+        })
+      .then(() => {
+        return request(app)
+          .post("/api/articles/1/comments/")
+          .send({
+            body: 'New comment to post',
+            username: 'icellusedkars'
+          },)
+          .expect(201)
+          .then(({ body }) => {
+              commentId = body.postedComment.comment_id
+            })
+      })
+      .then(() => {
+        return request(app)
+          .delete(`/api/comments/${commentId}`)
+          .expect(204);
+      })
+      .then(() => {
+        return request(app)
+          .get("/api/articles/1?comment_count=true")
+          .expect(200)})
+      .then(({ body }) => {
+        expect(body.article.comment_count).toBe(testedCommentCount);
+      });
+  });
+  test("DELETE 404: responds with error when comment_id does not exist", () => {
+    return request(app)
+      .delete("/api/comments/500")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Requested information not found!");
+      });
+  });
+  test("DELETE 400: responds with 'Bad Request!...' when comment_id entered in wrong format", () => {
+    return request(app)
+      .delete("/api/comments/wrong_format")
+      .expect(400)
+      .then(({ res }) => {
+        expect(res.statusMessage).toBe(
+          "Invalid request: not a valid integer entered!"
+        );
+      });
+  });
+});
```

Comment:
The tests for delete look good, again you don't need comment_count = true. Your first 204 test is unnecessarily complicated but it does work.